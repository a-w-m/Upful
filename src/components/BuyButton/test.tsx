import React from 'react'
import {render} from '@testing-library/react'
import BuyButton from "."

describe("BuyButton Component", ()=>{
    test("matches snapshot", ()=>{
        const {container} = render( <BuyButton
            className ='snipcart-add-item'
            data-item-id= ""
            data-item-price= ""
            data-item-name= ""
            data-item-description=""
            data-item-image=""
            data-item-url=""
            data-item-custom1-name=""
            data-item-custom1-options=""
            data-item-custom1-value="">
       </BuyButton>)
        expect(container.firstChild).toMatchSnapshot()

    })
})