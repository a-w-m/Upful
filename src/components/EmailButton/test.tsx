import React from "react"
import { render } from "@testing-library/react"
import EmailButton from "."

const props ={
    title: "test-email"
}

describe("EmailButton component", () => {
  test("matches snapshot", () => {
    const { container } = render(<EmailButton {...props}></EmailButton>)
    expect(container.firstChild).toMatchSnapshot()
  })


  test("component displays correct text", ()=>{
    const { queryByText } = render(<EmailButton {...props}></EmailButton>)
    expect(queryByText("Email us about this product")).toBeTruthy();

  })

  test("component has correct href", ()=>{
    const { getByRole } = render(<EmailButton {...props}></EmailButton>)

    // Click button
    expect(getByRole('link')).toHaveAttribute("href", `mailto:howdy@upful.com?Subject=${props.title}`)


  })

 
})
