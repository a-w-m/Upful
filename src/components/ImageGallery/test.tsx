import React from "react"
import { fireEvent, render, screen } from "@testing-library/react"
import ImageGallery from "./index"
import { imageGalleryData } from "../../../__mocks__/mock-data"

describe("ImageGallery Component", () => {
  const mockDispatch = jest.fn()

  test("matches snapshot", () => {
    const { container } = render(
      <ImageGallery
        images={imageGalleryData}
        dispatch={mockDispatch}
      ></ImageGallery>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test("dispatch is called when gallery image is clicked ", () => {
    render(
      <ImageGallery
        images={imageGalleryData}
        dispatch={mockDispatch}
      ></ImageGallery>
    )
    const image = screen.getAllByRole("listitem")[0]
    fireEvent.click(image)
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
})

test("no list rendered when no images supplied", () => {
  const { container } = render(
    <ImageGallery images={[]} dispatch={jest.fn()}></ImageGallery>
  )
  expect(container.firstChild).toBeNull()
})
