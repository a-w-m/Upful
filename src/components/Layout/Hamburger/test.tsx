import React from "react"
import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import Hamburger from "."

describe("Hamburger component", () => {
  const setOpen = jest.fn()

  it("should match snapshot", () => {
    render(<Hamburger open={false} setOpen={setOpen} />)
    const button = screen.getByRole("button")
    expect(button).toMatchSnapshot()
  })

  it("should handle click", async () => {
    render(<Hamburger open={false} setOpen={setOpen} />)
    const user = userEvent.setup()
    const button = screen.getByRole("button")
    await user.click(button)
    expect(setOpen).toHaveBeenCalledTimes(1)
  })

  it("should apply style when open", () => {
    render(<Hamburger open={true} setOpen={setOpen} />)
    const menu = document.querySelectorAll("span")
    expect(menu[0]).toHaveStyle({ transform: "rotate(45deg)" })
    expect(menu[1]).toHaveStyle({ opacity: "0", transform: "translateX(2rem)" })
    expect(menu[2]).toHaveStyle({ transform: "rotate(-45deg)" })
  })

  it("should apply style when not open", () => {
    render(<Hamburger open={false} setOpen={setOpen} />)
    const menu = document.querySelectorAll("span")
    expect(menu[0]).toHaveStyle({ transform: "rotate(0)" })
    expect(menu[1]).toHaveStyle({ opacity: "1", transform: "translateX(0)" })
    expect(menu[2]).toHaveStyle({ transform: "rotate(0)" })
  })
})
