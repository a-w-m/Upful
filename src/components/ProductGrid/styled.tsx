import styled from "styled-components"
import device from "../Layout/mediaQuery"

export const Container = styled.div`
  display: grid;
  width: auto;
  grid-template-columns: repeat(auto-fit, minmax(100px, 1fr));
  padding: 2rem 0;
  grid-gap: 2rem;
  justify-content: center;

@media ${device.laptop} {
  grid-template-columns: repeat(auto-fit, minmax(300px, .25fr));
  padding: 2rem 2rem;
}
`
