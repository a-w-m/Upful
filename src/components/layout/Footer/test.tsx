import React from 'react'
import {render, screen} from '@testing-library/react'
import Footer from "."


describe("Footer Component", ()=>{

    test("matches component", ()=>{
        const {container} = render(<Footer />)
        expect(container.firstElementChild).toMatchSnapshot()

    })

})