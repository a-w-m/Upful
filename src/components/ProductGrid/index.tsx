import React from "react"
import { C } from "../../interfaces"
import ProductThumbnail from "../ProductThumbnail"
import { Container } from "./styled"

const ProductGrid: React.FC<C.Grid> = props => {
  return (
    <Container>
      {props.edges.map((edge, index) => {
        const {
          id,
          title,
          price,
          thumbnail,
          galleryImages
        } = edge.node.childMarkdownRemark.frontmatter
        const { slug } = edge.node.childMarkdownRemark.fields

        return (
          <ProductThumbnail
            id = {id}
            title={title}
            price={price}
            thumbnail={thumbnail}
            slug={slug}
            key={index}
            galleryImages ={galleryImages}
          ></ProductThumbnail>
        )
      })}
    </Container>
  )
}
export default ProductGrid
