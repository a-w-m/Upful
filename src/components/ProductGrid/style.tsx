import styled from 'styled-components'

export const Container = styled.div`

	display: grid;
	width: auto;
	grid-auto-rows: auto;
	grid-template-columns: 1fr 1fr;
	padding: 2rem 0;
	grid-gap: 2.1rem;

`