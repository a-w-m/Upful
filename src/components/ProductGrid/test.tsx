import React from "react"
import { render, screen } from "src/utils/test/test-utils"
import ProductGrid from "./index"
import { getProductaGridData } from "src/utils/test/data/"

describe("ProductGrid", () => {
  const mockData = getProductaGridData()
  beforeEach(() => {
    render(<ProductGrid {...mockData} />)
  })

  it("should match snapshot", () => {
    expect(document.body).toMatchSnapshot()
  })

  it("should display all products passed as props", () => {
    const products = screen.getAllByRole("article")
    expect(products).toHaveLength(mockData.edges.length)
  })
})
