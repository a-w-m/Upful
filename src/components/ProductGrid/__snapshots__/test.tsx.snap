// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductGrid should match snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-jSMfEi OGIkm"
    >
      <article
        class="sc-bczRLJ bIIyW"
      >
        <a
          href="/easy-sweatpant-in-natural/"
        >
          <div
            class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
            data-gatsby-image-wrapper=""
            style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
          >
            <div
              style="max-width: 1030px; display: block;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,%3Csvg height='1030' width='1030' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                style="max-width: 100%; display: block; position: static;"
              />
            </div>
            <div
              aria-hidden="true"
              data-placeholder-image=""
              style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
            <img
              alt=""
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="/static/1a4732d2748806c6a0c45dcd0ac92ca4/a640e/image-1.webp"
              data-srcset="/static/1a4732d2748806c6a0c45dcd0ac92ca4/c7cc5/image-1.webp 258w,/static/1a4732d2748806c6a0c45dcd0ac92ca4/f26c5/image-1.webp 515w,/static/1a4732d2748806c6a0c45dcd0ac92ca4/a640e/image-1.webp 1030w"
              decoding="async"
              loading="lazy"
              sizes="(min-width: 1030px) 1030px, 100vw"
              style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
            />
            <noscript />
            <script
              type="module"
            >
              const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
            </script>
          </div>
        </a>
      </article>
      <article
        class="sc-bczRLJ bIIyW"
      >
        <a
          href="/haori-coat-in-washed-denim/"
        >
          <div
            class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
            data-gatsby-image-wrapper=""
            style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
          >
            <div
              style="max-width: 995px; display: block;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,%3Csvg height='995' width='995' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                style="max-width: 100%; display: block; position: static;"
              />
            </div>
            <div
              aria-hidden="true"
              data-placeholder-image=""
              style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
            <img
              alt=""
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="/static/5e566884fdc0f35caf38e43e3d3b3faa/87640/image-1.webp"
              data-srcset="/static/5e566884fdc0f35caf38e43e3d3b3faa/dc55b/image-1.webp 249w,/static/5e566884fdc0f35caf38e43e3d3b3faa/525ed/image-1.webp 498w,/static/5e566884fdc0f35caf38e43e3d3b3faa/87640/image-1.webp 995w"
              decoding="async"
              loading="lazy"
              sizes="(min-width: 995px) 995px, 100vw"
              style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
            />
            <noscript />
            <script
              type="module"
            >
              const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
            </script>
          </div>
        </a>
      </article>
      <article
        class="sc-bczRLJ bIIyW"
      >
        <a
          href="/mardi-dress-in-majorelle/"
        >
          <div
            class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
            data-gatsby-image-wrapper=""
            style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
          >
            <div
              style="max-width: 1989px; display: block;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,%3Csvg height='1989' width='1989' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                style="max-width: 100%; display: block; position: static;"
              />
            </div>
            <div
              aria-hidden="true"
              data-placeholder-image=""
              style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
            <img
              alt=""
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="/static/9bb4bfc8c3b86199d4c6f2ce7bd2fe68/e565f/image-1.webp"
              data-srcset="/static/9bb4bfc8c3b86199d4c6f2ce7bd2fe68/ce874/image-1.webp 497w,/static/9bb4bfc8c3b86199d4c6f2ce7bd2fe68/87640/image-1.webp 995w,/static/9bb4bfc8c3b86199d4c6f2ce7bd2fe68/e565f/image-1.webp 1989w"
              decoding="async"
              loading="lazy"
              sizes="(min-width: 1989px) 1989px, 100vw"
              style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
            />
            <noscript />
            <script
              type="module"
            >
              const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
            </script>
          </div>
        </a>
      </article>
      <article
        class="sc-bczRLJ bIIyW"
      >
        <a
          href="/mona-marble-dress/"
        >
          <div
            class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
            data-gatsby-image-wrapper=""
            style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
          >
            <div
              style="max-width: 832px; display: block;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,%3Csvg height='832' width='832' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                style="max-width: 100%; display: block; position: static;"
              />
            </div>
            <div
              aria-hidden="true"
              data-placeholder-image=""
              style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
            <img
              alt=""
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="/static/3ab4e7845d0eddea1b17acedc963dd31/8f66e/image-1.webp"
              data-srcset="/static/3ab4e7845d0eddea1b17acedc963dd31/c95dc/image-1.webp 208w,/static/3ab4e7845d0eddea1b17acedc963dd31/3f9a3/image-1.webp 416w,/static/3ab4e7845d0eddea1b17acedc963dd31/8f66e/image-1.webp 832w"
              decoding="async"
              loading="lazy"
              sizes="(min-width: 832px) 832px, 100vw"
              style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
            />
            <noscript />
            <script
              type="module"
            >
              const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
            </script>
          </div>
        </a>
      </article>
      <article
        class="sc-bczRLJ bIIyW"
      >
        <a
          href="/the-field-jumpsuit/"
        >
          <div
            class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
            data-gatsby-image-wrapper=""
            style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
          >
            <div
              style="max-width: 1024px; display: block;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,%3Csvg height='1024' width='1024' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                style="max-width: 100%; display: block; position: static;"
              />
            </div>
            <div
              aria-hidden="true"
              data-placeholder-image=""
              style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
            <img
              alt=""
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="/static/01ab4798217057a46bbd2c052b23d4bd/67ded/image-1.webp"
              data-srcset="/static/01ab4798217057a46bbd2c052b23d4bd/22bfc/image-1.webp 256w,/static/01ab4798217057a46bbd2c052b23d4bd/d689f/image-1.webp 512w,/static/01ab4798217057a46bbd2c052b23d4bd/67ded/image-1.webp 1024w"
              decoding="async"
              loading="lazy"
              sizes="(min-width: 1024px) 1024px, 100vw"
              style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
            />
            <noscript />
            <script
              type="module"
            >
              const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
            </script>
          </div>
        </a>
      </article>
      <article
        class="sc-bczRLJ bIIyW"
      >
        <a
          href="/virginia-dress/"
        >
          <div
            class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
            data-gatsby-image-wrapper=""
            style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
          >
            <div
              style="max-width: 2004px; display: block;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,%3Csvg height='2004' width='2004' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                style="max-width: 100%; display: block; position: static;"
              />
            </div>
            <div
              aria-hidden="true"
              data-placeholder-image=""
              style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
            <img
              alt=""
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="/static/9bc7109a20b6a0d06cfdc4e6bb8a3970/c7f64/image-1.webp"
              data-srcset="/static/9bc7109a20b6a0d06cfdc4e6bb8a3970/d9828/image-1.webp 501w,/static/9bc7109a20b6a0d06cfdc4e6bb8a3970/a5c80/image-1.webp 1002w,/static/9bc7109a20b6a0d06cfdc4e6bb8a3970/c7f64/image-1.webp 2004w"
              decoding="async"
              loading="lazy"
              sizes="(min-width: 2004px) 2004px, 100vw"
              style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
            />
            <noscript />
            <script
              type="module"
            >
              const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
            </script>
          </div>
        </a>
      </article>
      <article
        class="sc-bczRLJ bIIyW"
      >
        <a
          href="/peace-daisy-tee/"
        >
          <div
            class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
            data-gatsby-image-wrapper=""
            style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
          >
            <div
              style="max-width: 1000px; display: block;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,%3Csvg height='1000' width='1000' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                style="max-width: 100%; display: block; position: static;"
              />
            </div>
            <div
              aria-hidden="true"
              data-placeholder-image=""
              style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
            />
            <img
              alt=""
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="/static/63315f2eafa8f2947de0de1c5d3a32dd/3cd29/image-1.webp"
              data-srcset="/static/63315f2eafa8f2947de0de1c5d3a32dd/e7160/image-1.webp 250w,/static/63315f2eafa8f2947de0de1c5d3a32dd/5f169/image-1.webp 500w,/static/63315f2eafa8f2947de0de1c5d3a32dd/3cd29/image-1.webp 1000w"
              decoding="async"
              loading="lazy"
              sizes="(min-width: 1000px) 1000px, 100vw"
              style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
            />
            <noscript />
            <script
              type="module"
            >
              const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
            </script>
          </div>
        </a>
      </article>
    </div>
  </div>
</body>
`;
